autoscheduler/frontend/src/tests/AvailabilityUI.test.tsx

```diff
@@ -0,0 +1,267 @@
+      // ensures that all times are actually calculated based on mocked measurements
+      expect(queryByText(/NaN/)).not.toBeInTheDocument();
+      expect(queryByLabelText('Adjust Start Time'))
+        .toHaveAttribute('aria-valuetext', expectedStart);
+    });
+
+    test('with dragging', () => {
+      // arrange
+      const store = createStore(autoSchedulerReducer, applyMiddleware(thunk));
+      const {
+        getByText, getByLabelText, queryByText, queryByLabelText,
+      } = render(
+        <Provider store={store}>
+          <Schedule />
+        </Provider>,
+      );
+      const startEventProps = {
+        button: 0,
+        clientY: 500,
+      };
+      const endEventProps = {
+        button: 0,
+        clientY: 700,
+      };
+      const expectedStart = '14:30';
+      const expectedEnd = '17:10';
+      const dayZero = document.getElementsByClassName(styles.calendarDay)[0];
+      jest.spyOn(dayZero, 'clientHeight', 'get')
+        .mockImplementation(() => 1000);
+
+      // act
+      const monday = getByLabelText('Monday');
+      fireEvent.mouseDown(monday, startEventProps);
+      fireEvent.mouseMove(monday, endEventProps);
+      fireEvent.mouseUp(monday, endEventProps);
+
+      // assert
+      expect(getByText('BUSY')).toBeInTheDocument();
+      expect(queryByText(/NaN/)).not.toBeInTheDocument();
+      expect(queryByLabelText('Adjust Start Time'))
+        .toHaveAttribute('aria-valuetext', expectedStart);
+      expect(queryByLabelText('Adjust End Time'))
+        .toHaveAttribute('aria-valuetext', expectedEnd);
+    });
+
+    test('with an end time of 9 AM if dragged below that', () => {
+      // arrange
+      const store = createStore(autoSchedulerReducer, applyMiddleware(thunk));
+      const {
+        getByText, getByLabelText, queryByText, queryByLabelText,
+      } = render(
+        <Provider store={store}>
+          <Schedule />
+        </Provider>,
+      );
+      const startEventProps = {
+        button: 0,
+        clientY: 600,
+      };
+      const endEventProps = {
+        button: 0,
+        clientY: 1200,
+      };
+      const expectedStart = '14:30';
+      const expectedEnd = '21:00';
+      const dayZero = document.getElementsByClassName(styles.calendarDay)[0];
+      jest.spyOn(dayZero, 'clientHeight', 'get')
+        .mockImplementation(() => 1000);
+      dayZero.getBoundingClientRect = jest.fn<any, any>(() => ({
+        top: 100,
+        bottom: 1100,
+      }));
+
+      // act
+      const monday = getByLabelText('Monday');
+      fireEvent.mouseDown(monday, startEventProps);
+      fireEvent.mouseMove(monday, startEventProps);
+      fireEvent.mouseDown(queryByLabelText('Adjust End Time'), startEventProps);
+      fireEvent.mouseMove(monday, endEventProps);
+      fireEvent.mouseUp(monday, endEventProps);
+
+      // assert
+      expect(getByText('BUSY')).toBeInTheDocument();
+      expect(queryByText(/NaN/)).not.toBeInTheDocument();
```

Comment:
Really just for my understanding, but what does querying for `/NaN/` test for?

Reply:
Before I figured out how to properly mock mouse events, the start time and end time of each availability block would be `NaN`, so this makes sure that it's actually calculating the times using the mocked `MouseEvent`s thar I'm giving